lib/puppet/util/windows/user.rb

```diff
@@ -52,9 +52,20 @@ def check_token_membership
   module_function :check_token_membership
 
   def password_is?(name, password)
-    logon_user(name, password) { |token| }
-  rescue Puppet::Util::Windows::Error
-    false
+    # Windows API LogonUserW always disallows this with error 1327:
+    # Logon failure: user account restriction. Possible reasons are blank passwords
+    # not allowed, logon hour restrictions, or a policy restriction has been enforced.
+    if password.nil? || password == ''
+      raise Puppet::Util::Windows::Error.new(
+        "Windows does not have the ability to verify NULL or empty passwords"
+      )
+    end
```

Comment:
Its already moved and the tests were changed as well...

Discovered that there's no way to get a nil in there either (unless you call the code directly from a custom type and provider), so removed the special `nil` message.